go_backend/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis/v8"
+)
+
+var ctx = context.Background()
+
+var Rdb = redis.NewClient(&redis.Options{
+	Addr: "redis:6379",
+	DB:   0,
+})
+
+const (
+	PRODUCTION  = "prod"
+	DEVELOPMENT = "dev"
+)
+
+func main() {
+	environment := os.Getenv("APP_ENVIRONMENT")
+	streamchaserUrl := os.Getenv("HOST_NAME")
+
+	var origins []string
+
+	if environment == PRODUCTION {
+		origins = []string{fmt.Sprintf("http://%s", streamchaserUrl),
+			fmt.Sprintf("https://%s", streamchaserUrl)}
+	} else {
+		origins = []string{"*"}
+	}
```

Comment:
I don't think this is gonna cover the Vercel URL
You can use this regex(the one on master is actually broken, since it blocks https://streamchaser.vercel.app)

https://regexr.com/6ians
```regex
https:\/\/streamchaser.*.vercel.app
```

Reply:
Done